[tiif] iP

tiif opened the pull request:
# Duke Pro 

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ **SUPER FAST** to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

------------------------------------------------------------------------------------------

If you Java programmer, you can use it to practice Java too. Here's the ``main`` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tiif committed:
complete level 0

tiif committed:
complete level 0

tiif committed:
complete level 1

tiif committed:
complete level 2

tiif committed:
complete level-3

tiif committed:
completed level-4

tiif committed:
automated text ui testing

tiif committed:
add exception

tiif committed:
implemented delete

tiif committed:
complete Level-7

tiif committed:
Merge branch 'branch-Level-7'

tiif committed:
update Level-7

tiif committed:
Merge branch 'branch-Level-7'

tiif committed:
convert string to date object

tiif committed:
Merge branch 'branch-Level-8'

tiif committed:
refactor the code to be more oop

tiif committed:
transfrom other classes as packages

tiif committed:
Merge branch 'add-gradle-support' of https://github.com/tiif/ip

tiif committed:
bug fix for event and deadline

tiif committed:
add junit test

tiif committed:
create javadoc

tiif committed:
Merge branch 'A-JavaDoc'

tiif committed:
refactor for coding standard

tiif committed:
Merge branch 'branch-A-CodingStandard'

tiif committed:
Added search feature

tiif committed:
Merge branch 'branch-Level-9'

tiif committed:
setup checkstyle

tiif committed:
Merge branch 'branch-A-CheckStyle'

juzzztinsoong commented on src/main/java/duke/TaskList.java:
did u forget to add modifier for tasklist

juzzztinsoong commented on src/main/java/duke/Parser.java:
really long conditionals and commands, perhaps the functions could be more modular e.g. with a separate dateparser function

juzzztinsoong commented on src/main/java/duke/Event.java:
what if the assumption doesn't hold? :O

juzzztinsoong reviewed (commented):
great job overall, code seems functional with good quality, comments are comprehensive, although there are several changes I recommend:
Similar files could be grouped together into packages to make project easier to navigate (also helps with permissions)
Greater use of OOP - should the command parsing take place inside the Main class

sheryew commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+    protected String transformedDate;
+
+    /**
+     * Constructor for deadline class
```
For Javadoc comments, perhaps you can attempt:
1. Punctuation behind each parameter description.
2. Empty line between description and parameter section

sheryew commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,87 @@
+     */
+    public void run() throws IOException {
+        ui.printWelcomeMessage();
+        boolean isExit = false;
```
Hey, I like how your boolean variables are named to sound like booleans 👍

sheryew commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,59 @@
+        this.from = from;
+        this.to = to;
+        String parsed_from[] = from.split("\\s+");
+        String parsed_to[] = to.split("\\s+");
```
Perhaps a better way to instantiate arrays will be something like: 
String[] parsed_from = form.split("\\s+");

sheryew commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,188 @@
+     * run after parseCommand
+     * @return the index, in 0-indexed form
+     */
+    public int parseToIndex() {
```
Hey, I like how your names for the methods are verbs and written in camelCase 👍

sheryew commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,39 @@
+
+
+import org.junit.jupiter.api.Test;
+
```
Perhaps you could potentially group the two imports statements into the same paragraph?

sheryew commented:
Amazing job. I have added code specific comments as displayed above. There are certain errors that are repeated in other files but I didn't list it all out. Overall, great job 🥇

tiif committed:
Before code refactoring

tiif committed:
Added Gui for chatbot

tiif committed:
deleted some imports and comments

tiif committed:
Merge branch 'branch-Level-10'

tiif committed:
Update README

tiif committed:
Merge branch 'Use-GFMD'

tiif committed:
Create Assertion for Event, Parser and Storage class

There are certain assumptions that should hold at various point in the
code.

By creating assertion, we can make sure that the assumption still hold
when it is updated.

tiif committed:
Merge pull request #1 from tiif/branch-A-Assertion

Create Assertion for Event, Parser and Storage class

tiif committed:
Merge branch 'master' of https://github.com/tiif/ip

tiif committed:
Fix the checkstyle problem for every java file.

tiif committed:
Merge branch 'branch-A-CodeQuality'

tiif committed:
shorten the command

tiif committed:
Merge branch 'branch-C-FriendlierSyntax'

tiif committed:
update README

tiif committed:
Merge pull request #3 from tiif/branch-A-CodeQuality

Branch a code quality

tiif committed:
Merge branch 'master' of https://github.com/tiif/ip

tiif committed:
Merge branch 'master' into branch-A-Assertion

tiif committed:
update readme

tiif committed:
Merge pull request #4 from tiif/branch-A-Assertion

Branch a assertion

tiif committed:
Merge branch 'master' of https://github.com/tiif/ip

tiif committed:
Refactor the code to make it easier to read.
Apply SLAP to Parser.java and Duke.java to prevent long method.

tiif committed:
fixed file loading error

tiif committed:
fixed loading marked task issue

tiif committed:
refactor code

tiif committed:
add back Ui field in Duke

tiif committed:
Added User Guide

tiif committed:
Update user guide

tiif committed:
update build.gradle

tiif committed:
Move everything inside duke

tiif committed:
remove class files

tiif committed:
Refactor code to increase code quality

tiif committed:
Fixed find command

tiif committed:
Add guide for the find command

tiif committed:
fixed toString error

tiif committed:
Update README.md
